thriftbp/client_pool_test.go

```diff
@@ -60,6 +60,7 @@ func TestCustomClientPool(t *testing.T) {
 
 	if _, err := thriftbp.NewCustomClientPool(
 		thriftbp.ClientPoolConfig{
+			Addr:               ":9090",
```

Comment:
Actually this is a problem.

In `NewCustomClientPool` `Addr` is allowed to (and should) be empty, because in that function it uses `genAddr` arg instead of the `Addr` field from the config. So a single validation function for different usage is kinda weird.

We should only call `Validate` from `NewBaseplateClientPool`, and also mention it in `Validate`'s doc comment that it's only meant for `NewBaseplateClientPool`.

Reply:
Good catch,  that explains some of the behavior I've seen. My configuration didn't parse but the thrift client initialized just fine.

Reply:
Makes me thing whether that should be codified somehow through sub-types but maybe it's making it unnecessarily complex.

Reply:
I think we can revert this change now as you changed to only check `Validate` from `NewBaseplateClientPool`? we can probably add a new test to make sure that `NewBaseplateClientPool` checks `Validate` and returns config errors.